pipeline definition

littlejawa opened the pull request:
This commit provides our pipeline definitions, that will allow us to duplicate
our pipelines in order to support concurrent work on different versions of OSC,
while having CI ready for each branch.

Those files are not part of the build, and are not used by the Konflux pipelines
directly. We need to "apply" them from the CLI to our Konflux instance when we
want to duplicate our pipeline for a building from a new branch.
I am commiting those files in our repo to keep them available, so that we can do
that process again in the future, if need be.

The README.md file provides background information, and the steps we need to follow.

littlejawa committed:
pipeline definition: adding documentation and template files

This commit provides our pipeline definitions, that will allow us to duplicate
our pipelines in order to support concurrent work on different versions of OSC,
while having CI ready for each branch.

Those files are not part of the build, and are not used by the Konflux pipelines
directly. We need to "apply" them from the CLI to our Konflux instance when we
want to duplicate our pipeline for a building from a new branch.
I am commiting those files in our repo to keep them available, so that we can do
that process again in the future, if need be.

The README.md file provides background information, and the steps we need to follow.

Signed-off-by: Julien Ropé <jrope@redhat.com>

wainersm commented on pipelinedef/README.md:
```diff
@@ -0,0 +1,119 @@
+# Pipeline definition in Konflux
+
+While our main piepelines are building from the default branches of our various
```
s/piepelines/pipelines/

In the commit message as well.

wainersm commented on pipelinedef/README.md:
```diff
@@ -0,0 +1,119 @@
+- the "openshift-sandboxed-containers" application, and all its components
+- the "osc-test-catalog" application, and the associated "osc-test-fbc" component
+
+For each of them, we exctract the yaml definition found on the Konflux instance:
```
s/exctract/extract/

wainersm commented on pipelinedef/project.yaml:
```diff
@@ -0,0 +1,12 @@
+spec:
+  displayName: "This project is used to deploy sandboxed containers pipelines"
+  description: |
+    Most of the devalopment for Sandboxed Containers happen in the default branch
```
s/devalopment/development

gkurz commented:
> piepline definition: adding documentation and template files

typo in the commit titel :stuck_out_tongue_closed_eyes:

littlejawa committed:
pipeline def readme fixes

Signed-off-by: Julien Ropé <jrope@redhat.com>

wainersm commented:
Hi @littlejawa !

Thanks for the detailed information on README.md and this PR! It's amazing!

Let's suppose the 1.11.2 release...suppose that you configured konflux and applied the template... when I open Konflux's console, will I see an application called `openshift-sandboxed-containers-1.11.2`? or the new "devstream" is somehow configured within the current `openshift-sandboxed-containers`?

Also, could we save in templates the definition of osc-test-fbc-integration wich is the integration tests pipeline for the osc-test-catalog application?

littlejawa commented:
> Hi @littlejawa !
> 
> Thanks for the detailed information on README.md and this PR! It's amazing!
> 
> Let's suppose the 1.11.2 release...suppose that you configured konflux and applied the template... when I open Konflux's console, will I see an application called `openshift-sandboxed-containers-1.11.2`? or the new "devstream" is somehow configured within the current `openshift-sandboxed-containers`?
> 
> Also, could we save in templates the definition of osc-test-fbc-integration wich is the integration tests pipeline for the osc-test-catalog application?

Hey @wainersm 

All good questions :-)

I need to update this PR for 1.11.x, because we added new components. 
And yes, the new integration test needs to be duplicated too, so I will add it there.

For your first question: yes, a new application will be created. You don't see it because I deleted it already, but we used to have a "openshift-sandboxed-containers-v1-10" application, created by the template that you see in this PR.
When I update the template and apply it for 1.11, we will have "openshift-sandboxed-containers-v1-11".

Note that it doesn't include the "z" number in the application name - it's meant to build all future 1.11.x releases.

And another note: this will hopefully be the last release where we use this template, because it is deprecated, and Konflux is going to release a new (simpler?) way of managing multi-version applications. See https://issues.redhat.com/browse/KATA-4517 if you like to read long, boring details :-p

littlejawa committed:
pipelines: add the build-dm-verity-image-task

Our template did not duplicate the build task up to now, letting the branch
pipeline use the same task as the default pipeline.
This proved to be an issue during the 1.11 dev cycle, where the build task
was modified for 1.11, and made it impossible to build 1.10 dm-verity anymore.

To prevent that, we are now duplicating the build-dm-verity-image-task along with
everything else.

At this point it is unclear how publishing this separate task will happen.
We need to make sure the two versions of the task can be accessed separately.
Further work is probably needed, but at this point, this copy ensure we have separate
pipelines at least. Modifying the duplicated pipeline can happen at a later time
without impacting the default one.

Signed-off-by: Julien Ropé <jrope@redhat.com>

littlejawa committed:
pipelines: add the osc-dm-verity-image-debug

During the 1.11 development cycle, we introduced this component that builds
in parallel to the regular dm-verity image, but provide a version of the image
that simplifies debugging.

This is built in a separate application, so that we can be sure we're not
releasing this debug image by error.

Signed-off-by: Julien Ropé <jrope@redhat.com>

wainersm commented:
> > Hi @littlejawa !
> > Thanks for the detailed information on README.md and this PR! It's amazing!
> > Let's suppose the 1.11.2 release...suppose that you configured konflux and applied the template... when I open Konflux's console, will I see an application called `openshift-sandboxed-containers-1.11.2`? or the new "devstream" is somehow configured within the current `openshift-sandboxed-containers`?
> > Also, could we save in templates the definition of osc-test-fbc-integration wich is the integration tests pipeline for the osc-test-catalog application?
> 
> Hey @wainersm
> 
> All good questions :-)
> 
> I need to update this PR for 1.11.x, because we added new components. And yes, the new integration test needs to be duplicated too, so I will add it there.
> 
> For your first question: yes, a new application will be created. You don't see it because I deleted it already, but we used to have a "openshift-sandboxed-containers-v1-10" application, created by the template that you see in this PR. When I update the template and apply it for 1.11, we will have "openshift-sandboxed-containers-v1-11".
> 
> Note that it doesn't include the "z" number in the application name - it's meant to build all future 1.11.x releases.
>

It makes sense!
 
> And another note: this will hopefully be the last release where we use this template, because it is deprecated, and Konflux is going to release a new (simpler?) way of managing multi-version applications. See https://issues.redhat.com/browse/KATA-4517 if you like to read long, boring details :-p

Not that I like boring details, so I asked AI for a summary but unfortunately it could not fetch the comments and then I switch to the old approach: read it myself :) 

Thanks for the explanation!

littlejawa committed:
pipelines: add 1.11 components to the template

For our 1.11 release, we have added several components that need to be added
to the template.

This commit also adds a nudge reference from must-gather to the bundle, which
was added during the 1.11 dev.

Signed-off-by: Julien Ropé <jrope@redhat.com>

littlejawa committed:
pipeline: add the test-fbc integration test

Signed-off-by: Julien Ropé <jrope@redhat.com>

littlejawa commented:
/retest

wainersm commented:
Thanks for addressing my request (the integration pipeline)!

littlejawa committed:
pipeline: update the devstream.yaml file for 1.11

Signed-off-by: Julien Ropé <jrope@redhat.com>
